### ISSUE TITLE
Union was callable in 0.770 but is not in 0.780

### ISSUE THREAD

**Issue by feuGeneA** [python/mypy#8953]
Using version 0.770, I had some code passing type checking just fine, but now using 0.780 the same code yields `"object" not callable`. the code in question is the expression `Union[bytes, str](returned)`.

Did 0.780 change a `Union` from callable to non-callable? Or was it always non-callable and just not flagged before? Or what?

```bash session
 4Jun13:16:39 /$ mypy --version
mypy 0.770

 4Jun13:16:45 /$ mypy -c "from typing import Union; Union[str, bytes]('')"
Success: no issues found in 1 source file

 4Jun13:17:01 /$ pip install --upgrade mypy
Collecting mypy
  Using cached mypy-0.780-cp37-cp37m-manylinux1_x86_64.whl (20.2 MB)
Requirement already satisfied, skipping upgrade: typing-extensions>=3.7.4 in /home/gene/.pythonz/pythons/CPython-3.7.2/lib/python3.7/site-packages (from mypy) (3.7.4.2)
Requirement already satisfied, skipping upgrade: mypy-extensions<0.5.0,>=0.4.3 in /home/gene/.pythonz/pythons/CPython-3.7.2/lib/python3.7/site-packages (from mypy) (0.4.3)
Requirement already satisfied, skipping upgrade: typed-ast<1.5.0,>=1.4.0 in /home/gene/.pythonz/pythons/CPython-3.7.2/lib/python3.7/site-packages (from mypy) (1.4.1)
Installing collected packages: mypy
  Attempting uninstall: mypy
    Found existing installation: mypy 0.770
    Uninstalling mypy-0.770:
      Successfully uninstalled mypy-0.770
Successfully installed mypy-0.780

 4Jun13:17:29 /$ mypy -c "from typing import Union; Union[str, bytes]('')"
<string>:1: error: "object" not callable
Found 1 error in 1 file (checked 1 source file)
```

I checked the release notes for 0.780 and didn't see anything that _seemed_ relevant to me...

I'm not sure what behavior to expect.  I don't have that good of a grasp of Python type checking.  _Should_ a `Union` be callable, the way any other type like `int` or `str` are?  Or not? :man_shrugging: 

I'm using Python 3.7.something.  (3.7.2 locally, possibly a different version in my CI, where this is also happening.)

**Comment by JelleZijlstra** [comment]
Mypy generally doesn't like using types in a runtime context. `Union[bytes, str](returned)` doesn't work at runtime either, at least in 3.6 (it throws `TypeError: Cannot instantiate typing.Union`). I'm not sure what it could mean either.

**Comment by jstasiak** [comment]
Not knowing anything about the code using this I'd interpret it as equivalent to `cast(Union[bytes, str], returned)`, I'm curious how far off am I.

**Comment by hauntsaninja** [comment]
Yes, this was always not callable and is now flagged.
The relevant change was to typeshed, not mypy: https://github.com/python/typeshed/pull/3933.

Note this applies to simpler types like List as well. Type checkers will now correctly complain about code that fails at runtime like:
```
~/delete λ cat test89.py 
from typing import List
List()                                                                               
~/delete λ python3.9 test89.py
Traceback (most recent call last):
  File "/Users/shantanu/delete/test89.py", line 2, in <module>
    List()
  File "/Users/shantanu/.pyenv/versions/3.9-dev/lib/python3.9/typing.py", line 614, in __call__
    raise TypeError(f"Type {self._name} cannot be instantiated; "
TypeError: Type List cannot be instantiated; use list() instead
```

@jstasiak it's possible in theory mypy could learn those semantics, but it would need to be supported by CPython first so that it doesn't crash at runtime. I feel it's unlikely CPython will make that change, since `Union[bytes, str](x)` looks more like it's meant to do something at runtime than `cast(Union[bytes, str], x)`.

**Comment by jstasiak** [comment]
Oh sure, I'm not proposing Mypy and Python implement this, I'm just trying to guess the intended meaning of the code in the report.

**Comment by JukkaL** [comment]
There seems to be nothing to do here; mypy seems to be behaving now as expected, as a false negative was fixed in 0.780.
